distributed/worker.py

```diff
@@ -1396,18 +1396,18 @@ async def get_data(
     ###################
 
     def update_data(self, comm=None, data=None, report=True, serializers=None):
+        recommendations = {}
         for key, value in data.items():
             ts = self.tasks.get(key)
-            if getattr(ts, "state", None) is not None:
-                self.transition(ts, "memory", value=value)
-            else:
+            if ts is None:
                 self.tasks[key] = ts = TaskState(key)
-                self.put_key_in_memory(ts, value)
-                ts.priority = None
-                ts.duration = None
+                ts.state = "flight"
+            recommendations.update({ts: ("memory", {"value": value})})
```

Comment:
This hack was replacing the direct call to `put_key_in_memory` -- the `flight->memory` transition does the right thing, so I tagged it that way.  Gross.